sql/core/src/test/scala/org/apache/spark/sql/CollationSuite.scala

```diff
@@ -768,7 +768,7 @@ class CollationSuite extends DatasourceV2SQLBase with AdaptiveSparkPlanHelper {
     })
   }
 
-  test("hash based joins not allowed for non-binary collated strings") {
+  test("hash based joins are also allowed for non-binary collated strings") {
```

Comment:
Feel free to remove this test if `hash join should be used for collated strings` is enough.

In tests we shouldn't care about history - i.e. it is not important that once upon a time we didn't support hash joins and now we do. Tests should just assert that current behaviour is correct.